src/libs/merge/kdbmerge.c

```diff
@@ -609,6 +648,223 @@ static int checkSingleSet (KeySet * checkedSet, KeySet * firstCompared, KeySet *
 	return 0;
 }
 
+#ifdef LIBGITFOUND
+
+/**
+ * Iterates over all keys that belong to an key array started by arrayStart and puts their values into a newly allocated string
+ * Each value is separated by a newline.
+ * Does not modify arrayStart.
+ *
+ * @param ks the key set of which the array is part
+ * @param arrayStart key that is the start of the array /#0
+ * @param informationKey the key for error codes
+ * @returns pointer to the string, NULL on error
+ * */
+static char * getValuesAsArray (KeySet * ks, const Key * arrayStart, Key * informationKey)
+{
+	if (ks == NULL || arrayStart == NULL || informationKey == NULL)
+	{
+		ELEKTRA_SET_INTERNAL_ERROR (informationKey, "Parameters must not be null");
+		return NULL;
+	}
+	int size = 1; // size 1 for final \0
+	/**
+	 * this calloc sets the first character to \0 so that the first invocation of
+	 * strncat has something to append to
+	 */
+	char * result = elektraCalloc (1);
+	/** ksLookup does not get a copy of the key but the real key.
+	 *  The elektraArrayIncName would then change the name of the real key.
+	 *  We don't want that as we only increase the name to loop over all keys.
+	 */
+	Key * iterator = keyDup (arrayStart);
+	if (iterator == NULL)
+	{
+		ELEKTRA_SET_INTERNAL_ERROR (informationKey, "Could not duplicate key to iterate.");
+		elektraFree (result);
+		return NULL;
+	}
+	Key * lookup;
+	int counter = 0;
+	while ((lookup = ksLookup (ks, iterator, KDB_O_POP)) != 0)
+	{
+		counter++;
+		int tmpSize = keyGetValueSize (lookup);
+		size += tmpSize; // keyGetValueSize includes size for \0 which we use for \n instead
+		if (elektraRealloc ((void **) &result, size) < 0)
+		{
+			ELEKTRA_SET_OUT_OF_MEMORY_ERROR (informationKey, "Memory allocation failed.");
+			elektraFree (result);
+			keyDel (iterator);
+			return NULL;
+		}
+		strncat (result, keyString (lookup), size); // size of whole buffer
+		strcat (result, "\n");
+		if (counter >= 2)
+		{
+			int retval = keyDel (lookup);
+			if (retval != 0)
+			{
+				char msg[300];
+				if (retval < 0)
+				{
+					snprintf (msg, 300, "Could not delete key from key set because null pointer.");
+				}
+				else
+				{
+					snprintf (msg, 300, "Could not delete key with name %s from key set. There are %d references left.",
+						  keyName (lookup), retval);
+				}
+				ELEKTRA_SET_INTERNAL_ERROR (informationKey, msg);
+				elektraFree (result);
+				keyDel (iterator);
+				return NULL;
+			}
+		}
+		if (elektraArrayIncName (iterator) < 0)
+		{
+			ELEKTRA_SET_INTERNAL_ERROR (informationKey, "Increasing array key failed.");
+			elektraFree (result);
+			keyDel (iterator);
+			return NULL;
+		}
+	}
+	keyDel (iterator);
+	return result;
+}
+
+/**
+ * Turns an array from getValuesAsArray into a key set again.
+ * Frees the array and creates a new key set.
+ *
+ * @array array to be converted, has no \0 terminator at end
+ * @length length of array, does not include size for \0 as this is not there
+ * @informationKey for errors
```

Comment:
This does not seem to be valid doxygen. Please check the generated documentation.

Reply:
> This does not seem to be valid doxygen.

Correct.

> Please check the generated documentation.

However, I don't think this is possible. In contrast to `elektraMerge` I could not find something about this function in the generated documentation. I assume this is because it's static.